lib/ansible/module_utils/ipa.py

```diff
@@ -97,10 +97,14 @@ def _post_json(self, method, name, item=None):
             item = {}
         url = '%s/session/json' % self.get_base_url()
         data = dict(method=method)
-        if method != 'ping':
-            data['params'] = [[name], item]
-        else:
+
+        # TODO: We should probably handle this a little better.
```

Comment:
Could you please elaborate more about this ?

Reply:
@Akasurde So when I was adding this module, I found it surprising that I needed to make the change here in `module_utils/ipa.py`. I thought that _maybe_ we could do a better job with the function implementation so that we didn't have to add "exceptions"/"switch statements" for certain API methods like how we're doing now.

I honestly didn't think too much more about it than that, and just added that comment. :-)